### PR TITLE
*: support coverage for with-real-tikv

### LINKED ISSUE #44797: support coverage mode for with-real-tikv
## Enhancement

Currently, our with-real-tikv tests do not support coverage mode. We need to add support for coverage mode to these tests.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #44797

ref https://github.com/PingCAP-QE/ci/issues/2171

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: build/jenkins_collect_coverage.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (hawkingrei) [inline build/jenkins_collect_coverage.sh:1]: It is copied from the part of ```build/jenkins_unit_test.sh```

File: build/jenkins_collect_coverage.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Copyright 2022 PingCAP, Inc.
```
Comment (hawkingrei) [inline build/jenkins_collect_coverage.sh:3]:
```suggestion
# Copyright 2023 PingCAP, Inc.
```

**Review by xhebox — APPROVED** [review]

File: build/jenkins_collect_coverage.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test is used to test compatible for BR restore.
+# It will download backup data from internal file server.
+# And make sure these backup data can restore through newly BR tools to newly cluster.
+
+set -o pipefail
+
+bazel_collect
```
Comment (wuhuizuo) [inline build/jenkins_collect_coverage.sh:23]: it's better to add some ref repo link comment for this line.

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**